File: algorithm_problems/leetcode/001-050/007. Reverse_Integer.js
// https://leetcode.com/problems/reverse-integer/description/

var reverse = function(x) {
  let abs = Math.abs(x);
  let arr = [];

  while (abs >= 1) {
    arr.push(abs % 10);
    abs = Math.floor(abs / 10);
  }

  let result = 0;
  let tens = 1;
  while (arr.length) {
    result += arr.pop() * tens;
    tens *= 10;
  }

  const upper = Math.pow(2, 31) - 1;
  const lower = 0 - Math.pow(2, 31);

  if (result > upper || -result < lower) return 0;
  return x > 0 ? result : -result;
};

console.log(reverse(12531234));

File: algorithm_problems/leetcode/001-050/007. Reverse_Integer.py
# https://leetcode.com/problems/reverse-integer/description/

import math


class Solution:
    def reverse(self, x):
        """
      :type x: int
      :rtype: int
      """
        y = abs(x)
        arr = []

        while y >= 1:
            arr.append(y % 10)
            y = math.floor(y / 10)

        result = 0
        tens = 1
        while arr:
            result += arr.pop() * tens
            tens *= 10

        upper = pow(2, 31) - 1
        lower = 0 - pow(2, 31)

        if result > upper or -result < lower:
            return 0
        return result if x > 0 else -result

